q2cli/commands.py

```diff
@@ -217,15 +217,24 @@ def __call__(self, **kwargs):
                                               stderr=stderr):
                 results = action(**arguments)
         except Exception as e:
+            import traceback
             if verbose:
-                import traceback
                 import sys
                 traceback.print_exc(file=sys.stderr)
                 click.echo(err=True)
                 self._echo_plugin_error(e, 'See above for debug info.')
             else:
-                self._echo_plugin_error(
-                    e, 'Re-run with --verbose to see debug info.')
+                import tempfile
+                log_path = tempfile.NamedTemporaryFile(prefix='qiime2-err-',
+                                                       suffix='.log',
+                                                       delete=False)
+                with open(log_path.name, 'w+') as fh:
```

Comment:
This won't work on Windows (possibly other platforms) -- it is opening a second file handle to the temp file in `w+` mode (if that file even existed on the filesystem, which differs by platform). `log_path` is already a file-like object in `w+` mode.

I think we want to use a context manager to handle the temporary file object:

```python
with tempfile.NamedTemporaryFile(prefix='qiime2-q2cli-err-', suffix='.log', delete=False) as fh:
    traceback.print_exc(file=fh)
    log_path = fh.name

...
```

Also note that I modified the temp file prefix to include q2cli in the name, just to make it clear what q2 interface is creating the log.